### PR TITLE
Add Floki.traverse_and_update/2. Fix #206.

### LINKED ISSUE #206: Children/content manipulation support
Currently we can manipulate tag name and attributes with `Floki.map/2`, what about the children/content?

Use case is to override certain elements:
- Input: `picture` 
- Output: `<div>picture</div>`

### PR DESCRIPTION
**Pull request by ericlathrop**
Let me know if there's anything you want me to change/add.

### CODE REVIEW

**Comment by ericlathrop** [comment]
I had formatted with Elixir 1.8.1, and it looks like Travis wants formatting from an earlier version.

File: lib/floki.ex [hunk 1]
```diff
@@ -228,6 +228,22 @@ defmodule Floki do
...
+      iex> html = {"div", [], ["hello"]}
+      iex> Floki.traverse_and_update(html, fn {"div", attrs, children} -> {"p", attrs, children} end)
+      {"p", [], ["hello"]}
+
+      iex> html = {"div", [], [{"span", [], ["hello"]}]}
+      iex> Floki.traverse_and_update(html, fn {"span", _attrs, _children} -> nil; tag -> tag end)
+      {"div", [], []}
+  """
+
+  defdelegate traverse_and_update(html_tree, fun), to: Floki.Traversal
```
Comment (philss) [inline lib/floki.ex]: Nice!

File: lib/floki/traversal.ex [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  def traverse_and_update([], _fun), do: []
+
+  def traverse_and_update({elem, attrs, children}, fun) do
+    mapped_children = traverse_and_update(children, fun)
+    fun.({elem, attrs, mapped_children})
+  end
+
+  def traverse_and_update({:comment, children}, fun) do
```
Comment (philss) [inline lib/floki/traversal.ex]: we can have some other cases like `{:pi, _, _}` and `{:pi, _}`. So I think you could have a clause similar to the first one, but in the bottom.

```elixir
def traverse_and_update(html_node, _fun), do: html_node
```

Comment (philss) [reply lib/floki/traversal.ex]: I'm going to fix this after merge.

**Pull request merged**